### DISCUSSION TITLE
JSVM outside bindings potential for wifi mesh toy nodes / smart house / smart house

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#3282] [Show and tell]
I'm fine using the Pocketbase as a Golang framework for now.  I'm wondering what kind of potential there could be for having access to Raspberry / Banana Pi GPIO pins from Javascript.  Also Video4Linux pipes.

So I think that PB as framework is the way to go for a very serious project, but I have some non-serious projects in mind.  In particular I have a large homestead property that is all wired up with power now.  The next step is installing several Banana Pi router nodes to create a wifi mesh.  These devices are fairly powerful enough to run multiple cameras and a speedy Pocketbase web server.

So I'm mentally exploring the concept of something like HTMX build-less Pocketbase using JSVM.  It might just be a hell of a lot better and easier to not need to build and deploy to multiple nodes.  Some of these nodes could be pretty basic and not particularly important.  For something like this I could have a single git repo where the public directory for each node is just a directory in the repo that gets sent over or sent back after some on-the-fly tinkering.  I like the idea of being able to tinker on the live node because it then becomes something easier to work with, like one of PHP's advantages.

Also in support of this is [openvscode-server](https://github.com/gitpod-io/openvscode-server), which makes it super easy to develop right on the mesh node.  Then just access VS Code over network through the browser, and even install it as an web app if your desktop environment allows that.

I hope that you can think about this as you go forward.  Pocketbase could be a seriously good and magical solution for beginner smart home and smart farm automation.  I'm super happy with my professional Pocketbase as a framework setup, but it's just a lot of infrastructure overhead for some of the stuff that I want to be doing.

**Comment by whmeitzler** [comment]
GPIO can be handled by reading and writing files in the sysFS -/sys/class/gpio. If Pocketbase allows you to read and write files, and I'm pretty sure they do, you're off to the races.

Video looks to be wholly out of scope for Pocketbase. You'd be better served looking at ffmpeg, gstreamer, or other such tools.

Have fun with your gadgets, build something cool!

**Reply by dschissler** [reply]
> Video looks to be wholly out of scope for Pocketbase. You'd be better served looking at ffmpeg, gstreamer, or other such tools.

It's fine if it needed to rely on those utilities existing as system programs.  Well though, those v4l sockets would expose possibly already encoded video from the device.  The codecs could change per device.  It's likely that most encodings would be low effort and not particularly bit dense.
